libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +190,67 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam[] {
+    const settings = new Array<SetConfigurationSettingParam>();
+    const featureFlags = new Array<FeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
+    if (this.msFmFeatureManagementKeyWord) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaSection(config);
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} is in an invalid format: id is a required property`
+            );
+          }
+
+          if (!this.validateFeatureName(featureFlag.id)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature id.`
+            );
+          }
+          featureFlags.push(this.getFeatureFlagDefinitionFromMsFmSchema(featureFlag));
+        }
+      }
+    }
+
+    const legacySchemaFeatureFlags = this.getLegacySchema(config);
```

Comment:
Check "foundLegacySchema" before try to parse it.

Reply:
I have added a check before parsing it